### PR TITLE
A bunch of device changes

### PR DESCRIPTION
**Pull request by pazos**
Move most device independant stuff to activityExtensions.
Make both `device.epd` and `device.lights` public, so their methods can be accesed from `MainActivity`(removing a lot of duplication)
Decouple lights dialog from its implementation. That's required to have a `LightsTestActivity`without duplicating code.

@zwim: could you test if the frontlight widget behaves ok with this PR in?. I don't have access to any android eink device right now.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/android-luajit-launcher/339)

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]
lgtm, I'm fine with before and after though. :-)

Btw, let's wait until next week to merge, at least in front.

**Comment by zwim** [comment]
@pazos Unfortunately I am not at home until saturday.  I will look at it on sunday.

File: app/src/main/java/org/koreader/launcher/MainActivity.kt [hunk 1]
```diff
@@ -406,34 +439,34 @@ class MainActivity : NativeActivity(), LuaInterface,
...
     }
 
     override fun getScreenMinBrightness(): Int {
-        return device.getScreenMinBrightness()
+        return device.lights.getMinBrightness()
     }
 
     override fun getScreenMaxWarmth(): Int {
-        return device.getScreenMaxWarmth()
+        return device.lights.getMaxBrightness()
```
[the remark this answers is not in the document]
Comment (pazos) [reply app/src/main/java/org/koreader/launcher/MainActivity.kt:450]: Thanks!

**Comment by pazos** [comment]
Fixed the typo and added an unified test activity for epd/lights.

A few things are missing, namely:

1. The link to the gh issue isn't clickable yet.
2. The generated report is useless.

Anyways, it should be enough for @zwim to test if both dropdown+button methods work fine with a working driver.

Also, language suggestions are very welcome @Frenzie :)

Attached a screenshot:
test

**Comment by Frenzie** [comment]
You forgot a period at the end of the third sentence but all sounds good to me! :+1: 

Possibly something like "try now" or "test" instead?

**Comment by zwim** [comment]
@pazos I will look into it tomorrow.

**Comment by zwim** [comment]
@pazos: Tested on a Tolino Freescale/NTX:
1.) Lights dropdown: For all devices I get a working dialog (but of course only on Tolino brightness and warmth actually change)

2.) E-Ink dropdown: I can select 4 entries, but if I select and then try any of them I don't notice any changes, but `adb logcat` shows me (for all 4 entries):
```
I/TestActivity( 8463): running epd test: Onyx/Qualcomm
W/System.err( 8463): java.lang.NullPointerException: v must not be null
W/System.err( 8463): 	at org.koreader.launcher.TestActivity.runEpd(TestActivity.kt:12)
W/System.err( 8463): 	at org.koreader.launcher.TestActivity.onCreate$lambda-1(TestActivity.kt:1)
W/System.err( 8463): 	at org.koreader.launcher.TestActivity.lambda$DUPziKZdqnyUHBHCou4p9pSDX1A(TestActivity.kt)
W/System.err( 8463): 	at org.koreader.launcher.-$$Lambda$TestActivity$DUPziKZdqnyUHBHCou4p9pSDX1A.onClick(lambda)
W/System.err( 8463): 	at android.view.View.performClick(View.java:4666)
W/System.err( 8463): 	at android.view.View$PerformClick.run(View.java:18925)
W/System.err( 8463): 	at android.os.Handler.handleCallback(Handler.java:733)
W/System.err( 8463): 	at android.os.Handler.dispatchMessage(Handler.java:95)
W/System.err( 8463): 	at android.os.Looper.loop(Looper.java:136)
W/System.err( 8463): 	at android.app.ActivityThread.main(ActivityThread.java:5017)
W/System.err( 8463): 	at java.lang.reflect.Method.invokeNative(Native Method)
W/System.err( 8463): 	at java.lang.reflect.Method.invoke(Method.java:515)
W/System.err( 8463): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
W/System.err( 8463): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
W/System.err( 8463): 	at dalvik.system.NativeStart.main(Native Method)
```

3.) below that the correct info is shown:
``` 
Device

Manufacturer: rakuten kobo inc.
Brand: rakutenkobo
Model: tolino
Device: ntx_6sl
...
```
4.) The three lines with text are really on the left edge of the screen. Maybe you can put them a bit to the right?

**Comment by pazos** [comment]
@zwim: thanks a lot for your tests!!

Could you check if this change fixes the issue with the EPD test?

```diff
diff --git a/app/src/main/java/org/koreader/launcher/TestActivity.kt b/app/src/main/java/org/koreader/launcher/TestActivity.kt
index 8122648..6914ac1 100644
--- a/app/src/main/java/org/koreader/launcher/TestActivity.kt
+++ b/app/src/main/java/org/koreader/launcher/TestActivity.kt
@@ -120,7 +120,8 @@ class TestActivity: AppCompatActivity() {
     private fun runEpd(id: String) {
         try {
             Log.i(tag, String.format("running epd test: %s", id))
-            val v = window.decorView.findViewById<View>(R.id.content)
+            //val v = window.decorView.findViewById<View>(R.id.content)
+            val v = binding.root
             epdMap[id]?.let { driver ->
                 when (id) {
                     "Freescale/NTX",
```

**Comment by zwim** [comment]
For Freescale/NTX the screen (_below the Title_ `Test`) does some kind of refresh.
The other entries do nothing.

But notice the `NullPointerExcepion` on `running epd test: Rockchip RK3026`:

```

I/TestActivity( 8820): running epd test: Freescale/NTX
I/EPD     ( 8820): requested eink refresh, type: 34 x:0 y:0 w:1072 h:1313

W/InputMethodManagerService( 2503): Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@41c7ef30 attribute=null, token = android.os.BinderProxy@41f72188
I/TestActivity( 8820): running epd test: Rockchip RK3368
E/EPD     ( 8820): java.lang.NoSuchMethodException: setByEinkUpdateMode [int]

W/InputMethodManagerService( 2503): Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@41f44bf0 attribute=null, token = android.os.BinderProxy@41f72188
I/TestActivity( 8820): running epd test: Onyx/Qualcomm
E/EPD     ( 8820): java.lang.NoSuchMethodException: setWaveformAndScheme [int, int, int]

W/InputMethodManagerService( 2503): Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@41d59500 attribute=null, token = android.os.BinderProxy@41f72188
I/TestActivity( 8820): running epd test: Rockchip RK3026
W/System.err( 8820): java.lang.NullPointerException
W/System.err( 8820): 	at org.koreader.launcher.device.epd.rockchip.RK30xxEPDController$Companion.requestEpdMode(RK30xxEPDController.kt:1)
W/System.err( 8820): 	at org.koreader.launcher.device.epd.RK3026EPDController.setEpdMode(RK3026EPDController.kt:1)
W/System.err( 8820): 	at org.koreader.launcher.TestActivity.runEpd(TestActivity.kt:5)
W/System.err( 8820): 	at org.koreader.launcher.TestActivity.onCreate$lambda-1(TestActivity.kt:1)
W/System.err( 8820): 	at org.koreader.launcher.TestActivity.lambda$DUPziKZdqnyUHBHCou4p9pSDX1A(TestActivity.kt)
W/System.err( 8820): 	at org.koreader.launcher.-$$Lambda$TestActivity$DUPziKZdqnyUHBHCou4p9pSDX1A.onClick(lambda)
W/System.err( 8820): 	at android.view.View.performClick(View.java:4666)
W/System.err( 8820): 	at android.view.View$PerformClick.run(View.java:18925)
W/System.err( 8820): 	at android.os.Handler.handleCallback(Handler.java:733)
W/System.err( 8820): 	at android.os.Handler.dispatchMessage(Handler.java:95)
W/System.err( 8820): 	at android.os.Looper.loop(Looper.java:136)
W/System.err( 8820): 	at android.app.ActivityThread.main(ActivityThread.java:5017)
W/System.err( 8820): 	at java.lang.reflect.Method.invokeNative(Native Method)
W/System.err( 8820): 	at java.lang.reflect.Method.invoke(Method.java:515)
W/System.err( 8820): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
W/System.err( 8820): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
W/System.err( 8820): 	at dalvik.system.NativeStart.main(Native Method)
```

**Comment by pazos** [comment]
> For Freescale/NTX the screen (below the Title Test) does some kind of refresh.
The other entries do nothing.

It is expected. The view we're trying to refresh doesn't contain the title, but the layout described in the new file `test.xml`. Here's the preview from android studio:
screenshot

If that is refreshed/updated we're fine :)

> But notice the NullPointerExcepion on running epd test: Rockchip RK3026

Also expected. We're printing the full stacktrace in the case of an exception, which I see in all calls except for the Freescale/NTX one. It just happens that the `RK3026` has a bigger stacktrace than `RK3368` or `Onyx/Qualcomm`

**Comment by pazos** [comment]
Almost done, everything should work fine. It finally looks like
screenshot


Feel free to test if you wish.
The only thing left is to open a new ticket and update the link .

**Comment by zwim** [comment]
Looks good so far.

**Review by Galunid — COMMENTED** [review]
Tested on the Onyx Nova 2 and lights/warmth/refresh work fine. Test seems to work ok too. It's probably a good idea to rename it in the menu from "Start eink test" to something more general. Let me know if you want anything else tested

**Pull request merged**